common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -898,7 +898,16 @@ bool network_utils::arp_send(const std::string &iface, const std::string &dst_ip
     struct sockaddr_ll sock;
     uint8_t packet_buffer[128];
 
-    uint32_t dst_ip_uint = network_utils::uint_ipv4_from_string(dst_ip);
+    // If the destination IP is empty, there is no point sending the arp, therefore replace the
```

Comment:
I just don't understand the comment "there is no point sending the arp" - you still do send the arp and there is a very good point. I think the comment should be updated to reflect the reason for why the destination IP may be empty.
Also please explain in the comment which clients were tested and answers broadcast ARPs.

Reply:
There is no point sending the arp since any client which will receive it will ignore it because the request has ip of 0.0.0.0.
The reason that the IP is empty is irrelevant for this commit. This commit makes the function robust.

Reply:
I believe the real issue is that with IP address 0.0.0.0, the client will ignore it - as mentioned in the commit message.

But actually, this makes me think: why do we ever need an IP address? AFAIU the only reason we have these IP addresses is to be able to do this ARP, but the IP addresses are non-standard. So if we can simply use the broadcast address, we really don't ever need an IP, and we can always use the broadcast address.

Probably not for this PR though.

Reply:
I agree with @tomereli's first point - The comment is indeed not clear enough.
You can write something similar to: "If the STA's IP address is unknown, instead of sending an empty value (0.0.0.0), a broadcast address is used (255.255.255.255)".

As for the second part, I don't see any reason to list any clients at this point. 
The new implementation is much better than the old one, even if not all clients respond to broadcast ARP requests.
It was tested already with several devices, but in the following couple of weeks will be tested with many more.